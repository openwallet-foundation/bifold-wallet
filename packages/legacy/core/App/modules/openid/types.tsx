import {
  OpenId4VciCredentialSupported,
  OpenId4VciIssuerMetadataDisplay,
  OpenId4VcSiopResolvedAuthorizationRequest,
} from '@credo-ts/openid4vc'
import { CredentialMetadata } from './display'
import { ClaimFormat, DifPexCredentialsForRequest } from '@credo-ts/core'

export type CredentialForDisplayId = `w3c-credential-${string}` | `sd-jwt-vc-${string}`
export interface OpenId4VcCredentialMetadata {
  credential: {
    display?: OpenId4VciCredentialSupported['display']
    order?: OpenId4VciCredentialSupported['order']
  }
  issuer: {
    display?: OpenId4VciIssuerMetadataDisplay[]
    id: string
  }
}

export type W3cIssuerJson = {
  id: string
}

export type W3cCredentialSubjectJson = {
  id?: string
  [key: string]: unknown
}

export type W3cCredentialJson = {
  type: Array<string>
  issuer: W3cIssuerJson
  issuanceDate: string
  expiryDate?: string
  credentialSubject: W3cCredentialSubjectJson | W3cCredentialSubjectJson[]
}

export type JffW3cCredentialJson = W3cCredentialJson & {
  name?: string
  description?: string
  credentialBranding?: {
    backgroundColor?: string
  }

  issuer:
    | string
    | (W3cIssuerJson & {
        name?: string
        iconUrl?: string
        logoUrl?: string
        image?: string | { id?: string; type?: 'Image' }
      })
}

export interface DisplayImage {
  url?: string
  altText?: string
}

export interface CredentialDisplay {
  name: string
  issuer: CredentialIssuerDisplay
  locale?: string
  description?: string
  textColor?: string
  backgroundColor?: string
  backgroundImage?: DisplayImage
  logo?: DisplayImage
  primary_overlay_attribute?: string
}

export interface CredentialIssuerDisplay {
  name: string
  locale?: string
  logo?: DisplayImage
}

export interface W3cCredentialDisplay {
  id: string
  createdAt: Date
  display: CredentialDisplay
  credential?: W3cCredentialJson
  attributes: W3cCredentialSubjectJson
  metadata: CredentialMetadata
  claimFormat: ClaimFormat
}

export interface OpenId4VPRequestRecord extends OpenId4VcSiopResolvedAuthorizationRequest {
  verifierHostName: string | undefined
  createdAt: string | Date
  credentialsForRequest: DifPexCredentialsForRequest | undefined
  type: string
}
