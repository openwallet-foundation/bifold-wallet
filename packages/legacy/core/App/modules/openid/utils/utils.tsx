import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}
