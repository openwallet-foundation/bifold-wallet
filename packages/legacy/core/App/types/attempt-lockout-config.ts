export type AttemptLockoutConfig = {
  baseRules: Record<number, number | undefined>
  thresholdRules: {
    threshold: number
    increment: number
    thresholdPenaltyDuration: number
  }
}
