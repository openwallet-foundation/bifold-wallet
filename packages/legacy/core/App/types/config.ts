import { Locales } from '../localization'
import { ContactDetailsOptionsParams } from './contact-details'
import { PINSecurityParams } from './security'
import { SettingSection } from './settings'
import { Agent } from '@credo-ts/core'
import { StackNavigationOptions } from '@react-navigation/stack'
interface PushNotificationConfiguration {
  // function to get the current push notification permission status
  status: () => Promise<'denied' | 'granted' | 'unknown'>
  // function to request permission for push notifications
  setup: () => Promise<'denied' | 'granted' | 'unknown'>
  //function to call when the user changes the push notification setting
  toggle: (state: boolean, agent: Agent<any>) => Promise<void>
}
export interface Config {
  PINSecurity: PINSecurityParams
  proofTemplateBaseUrl?: string
  settings: SettingSection[]
  supportedLanguages: Locales[]
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  enableChat?: boolean
  enableTours?: boolean
  enableImplicitInvitations?: boolean
  enableReuseConnections?: boolean
  enableHiddenDevModeTrigger?: boolean
  showPreface?: boolean
  showPINExplainer?: boolean
  disableOnboardingSkip?: boolean
  enablePushNotifications?: PushNotificationConfiguration
  whereToUseWalletUrl?: string
  showScanHelp?: boolean
  showScanButton?: boolean
  showScanErrorButton?: boolean
  globalScreenOptions?: StackNavigationOptions
  showDetailsInfo?: boolean
  contactHideList?: string[]
  contactDetailsOptions?: ContactDetailsOptionsParams
  credentialHideList?: string[]
  disableContactsInSettings?: boolean
  disableMediatorCheck?: boolean
  internetReachabilityUrls: string[]
}

export interface HistoryEventsLoggerConfig {
  logAttestationAccepted: boolean
  logAttestationRefused: boolean
  logAttestationRemoved: boolean
  logInformationSent: boolean
  logInformationNotSent: boolean
  logConnection: boolean
  logConnectionRemoved: boolean
  logAttestationRevoked: boolean
  logPinChanged: boolean
  logToggleBiometry: boolean
}
