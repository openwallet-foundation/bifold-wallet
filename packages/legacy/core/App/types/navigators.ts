import { SdJwtVcRecord, W3cCredentialRecord } from '@credo-ts/core'
import { NavigatorScreenParams } from '@react-navigation/native'
import { StackNavigationOptions } from '@react-navigation/stack'
import { OpenId4VPRequestRecord } from '../modules/openid/types'
import { LayoutProps } from '../layout/ScreenLayout'

export enum Screens {
  AttemptLockout = 'Temporarily Locked',
  Splash = 'Splash',
  Onboarding = 'Onboarding',
  Terms = 'Terms',
  Preface = 'Preface',
  CreatePIN = 'Create a PIN',
  EnterPIN = 'Enter PIN',
  Home = 'Home',
  Scan = 'Scan',
  PasteUrl = 'Paste URL',
  Credentials = 'Credentials',
  CredentialDetails = 'Credential Details',
  CredentialOffer = 'Credential Offer',
  OpenIDCredentialDetails = 'Open ID Credential details',
  OpenIDCredentialOffer = 'Open ID Credential offer',
  OpenIDProofPresentation = 'Open ID Proof Presentation',
  ProofRequest = 'Proof Request',
  ProofRequestDetails = 'Proof Request Details',
  ProofRequestUsageHistory = 'Proof Request Usage History',
  Settings = 'Settings',
  Language = 'Language',
  DataRetention = 'Data Retention',
  Tours = 'Tours',
  Contacts = 'Contacts',
  ContactDetails = 'Contact Details',
  WhatAreContacts = 'What Are Contacts',
  Chat = 'Chat',
  Connection = 'Connection',
  MobileVerifierLoading = 'Mobile Verifier Loading',
  OnTheWay = 'On The Way',
  Declined = 'Declined',
  UseBiometry = 'Use Biometry',
  UsePushNotifications = 'Use Push Notifications',
  Developer = 'Developer',
  CustomNotification = 'Custom Notification',
  ProofChangeCredential = 'Choose a credential',
  ProofRequests = 'Proof Requests',
  ProofRequesting = 'Proof Requesting',
  ProofDetails = 'Proof Details',
  NameWallet = 'Name Wallet',
  RenameContact = 'Rename Contact',
  ScanHelp = 'Scan Help',
  HistorySettings = 'History Settings',
  HistoryPage = 'History',
  HistoryDetails = 'History details',
  AutoLock = 'AutoLock',
}

export enum Stacks {
  TabStack = 'Tab Stack',
  HomeStack = 'Home Stack',
  ConnectStack = 'Connect Stack',
  CredentialStack = 'Credentials Stack',
  SettingStack = 'Settings Stack',
  ContactStack = 'Contacts Stack',
  ProofRequestsStack = 'Proof Requests Stack',
  NotificationStack = 'Notifications Stack',
  ConnectionStack = 'Connection Stack',
  HistoryStack = 'History Stack',
  CustomNavStack1 = 'Custom Nav Stack 1',
}

export enum TabStacks {
  HomeStack = 'Tab Home Stack',
  ConnectStack = 'Tab Connect Stack',
  CredentialStack = 'Tab Credential Stack',
}

export type RootStackParams = {
  [Screens.Splash]: undefined
  [Stacks.TabStack]: NavigatorScreenParams<TabStackParams>
  [Screens.Chat]: { connectionId: string }
  [Stacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [Stacks.SettingStack]: NavigatorScreenParams<SettingStackParams>
  [Stacks.ContactStack]: NavigatorScreenParams<ContactStackParams>
  [Stacks.ProofRequestsStack]: NavigatorScreenParams<ProofRequestsStackParams>
  [Stacks.NotificationStack]: NavigatorScreenParams<NotificationStackParams>
  [Stacks.HistoryStack]: NavigatorScreenParams<HistoryStackParams>
  [Stacks.CustomNavStack1]: undefined
}

export type TabStackParams = {
  [TabStacks.HomeStack]: NavigatorScreenParams<HomeStackParams>
  [TabStacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [TabStacks.CredentialStack]: NavigatorScreenParams<CredentialStackParams>
}

export type AuthenticateStackParams = {
  [Screens.Preface]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Terms]: undefined
  [Screens.AttemptLockout]: undefined
  [Screens.CreatePIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.EnterPIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.UseBiometry]: undefined
  [Screens.NameWallet]: undefined
  [Screens.UsePushNotifications]: undefined
}

export type OnboardingStackParams = {
  [Screens.Onboarding]: undefined
  [Screens.Developer]: undefined
}

export type ContactStackParams = {
  [Screens.Contacts]: undefined
  [Screens.Chat]: { connectionId: string }
  [Screens.ContactDetails]: { connectionId: string }
  [Screens.RenameContact]: { connectionId: string }
  [Screens.WhatAreContacts]: undefined
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofDetails]: { recordId: string; isHistory?: boolean }
  [Screens.ProofRequest]: { proofId: string }
}

export type ProofRequestsStackParams = {
  [Screens.ProofRequests]: { connectionId?: string }
  [Screens.ProofRequesting]: { templateId: string; predicateValues?: Record<string, Record<string, number>> }
  [Screens.ProofDetails]: { recordId: string; isHistory?: boolean; senderReview?: boolean }
  [Screens.ProofRequestDetails]: { templateId: string; connectionId?: string }
  [Screens.MobileVerifierLoading]: { proofId: string; connectionId: string }
  [Screens.ProofRequestUsageHistory]: { templateId: string }
  [Screens.ProofChangeCredential]: {
    selectedCred: string
    altCredentials: string[]
    proofId: string
    onCredChange: (arg: string) => void
  }
}

export type CredentialStackParams = {
  [Screens.Credentials]: undefined
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.OpenIDCredentialDetails]: { credentialId: string }
}

export type HomeStackParams = {
  [Screens.Home]: undefined
}

export type ConnectStackParams = {
  [Screens.Scan]: undefined
  [Screens.NameWallet]: undefined
  [Screens.ScanHelp]: undefined
  [Screens.PasteUrl]: undefined
}

export type SettingStackParams = {
  [Screens.NameWallet]: undefined
  [Screens.Settings]: undefined
  [Screens.Language]: undefined
  [Screens.DataRetention]: undefined
  [Screens.Tours]: undefined
  [Screens.UseBiometry]: undefined
  [Screens.CreatePIN]: undefined
  [Screens.Terms]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Developer]: undefined
  [Screens.UsePushNotifications]: { isMenu?: boolean }
  [Screens.HistorySettings]: undefined
  [Screens.AutoLock]: undefined
}

export type NotificationStackParams = {
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.OpenIDCredentialDetails]: {
    credential: SdJwtVcRecord | W3cCredentialRecord
  }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.CustomNotification]: undefined
  [Screens.ProofDetails]: { recordId: string }
}

export type DeliveryStackParams = {
  [Screens.Connection]: {
    oobRecordId?: string
    proofId?: string
    credentialId?: string
    openIDUri?: string
    openIDPresentationUri?: string
  }
  [Screens.MobileVerifierLoading]: { proofId: string; connectionId: string }
  [Screens.ProofDetails]: { recordId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.OnTheWay]: { credentialId: string }
  [Screens.Declined]: { credentialId: string }
  [Screens.Chat]: { connectionId: string }
  [Screens.OpenIDCredentialOffer]: {
    credential: SdJwtVcRecord | W3cCredentialRecord
  }
  [Screens.OpenIDProofPresentation]: { credential: OpenId4VPRequestRecord }
}

export type HistoryStackParams = {
  [Screens.HistoryPage]: undefined
}

export type ScreenLayoutConfig = Partial<Record<Screens, LayoutProps>>
export type ScreenOptionsType = Partial<Record<Screens, StackNavigationOptions>>
