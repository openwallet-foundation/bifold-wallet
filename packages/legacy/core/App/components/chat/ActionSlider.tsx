import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

import { hitSlop } from '../../constants'
import { useTheme } from '../../contexts/theme'
import { testIdWithKey } from '../../utils/testable'
import SafeAreaModal from '../modals/SafeAreaModal'

interface Action {
  text: string
  icon: React.FC
  onPress: () => void
}

interface Props {
  actions: Action[] | undefined
  onDismiss: () => void
}

const ActionSlider: React.FC<Props> = ({ actions, onDismiss }) => {
  const { ColorPallet, TextTheme } = useTheme()
  const { t } = useTranslation()

  const styles = StyleSheet.create({
    centeredView: {
      marginTop: 'auto',
      justifyContent: 'flex-end',
    },
    outsideListener: {
      height: '100%',
    },
    modalView: {
      backgroundColor: ColorPallet.grayscale.white,
      borderTopStartRadius: 20,
      borderTopEndRadius: 20,
      shadowColor: '#000',
      padding: 20,
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 4,
      elevation: 5,
    },
    drawerTitleText: {
      ...TextTheme.bold,
      textAlign: 'center',
      marginVertical: 10,
    },
    drawerContentText: {
      ...TextTheme.normal,
    },
    drawerRow: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center',
      marginVertical: 12,
    },
    drawerRowItem: {
      color: ColorPallet.grayscale.black,
    },
  })

  return (
    <SafeAreaModal animationType="slide" transparent={true} onRequestClose={onDismiss}>
      <TouchableOpacity style={styles.outsideListener} onPress={onDismiss} />
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <TouchableOpacity
            testID={testIdWithKey('Close')}
            accessibilityLabel={t('Global.Close')}
            accessibilityRole={'button'}
            onPress={onDismiss}
            hitSlop={hitSlop}
            style={{ alignSelf: 'flex-end' }}
          >
            <Icon name="window-close" size={35} style={styles.drawerRowItem}></Icon>
          </TouchableOpacity>
          {actions &&
            actions.map((action) => {
              return (
                <TouchableOpacity
                  key={action.text}
                  testID={testIdWithKey(action.text)}
                  accessibilityLabel={testIdWithKey(action.text)}
                  accessibilityRole="button"
                  style={styles.drawerRow}
                  onPress={action.onPress}
                >
                  <action.icon />
                  <Text style={{ ...styles.drawerRowItem, marginLeft: 5 }}>{action.text}</Text>
                </TouchableOpacity>
              )
            })}
        </View>
      </View>
    </SafeAreaModal>
  )
}

export default ActionSlider
