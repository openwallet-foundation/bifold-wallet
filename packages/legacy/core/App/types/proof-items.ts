import { AnonCredsCredentialsForProofRequest } from '@credo-ts/anoncreds'
import { CredentialExchangeRecord } from '@credo-ts/core'
import { Attribute, Predicate } from '@hyperledger/aries-oca/build/legacy'
import { DescriptorMetadata } from '../utils/anonCredsProofRequestMapper'

export type CredentialDataForProof = {
  groupedProof: (ProofCredentialPredicates & ProofCredentialAttributes)[]
  retrievedCredentials: AnonCredsCredentialsForProofRequest | undefined
  fullCredentials: CredentialExchangeRecord[]
  descriptorMetadata?: DescriptorMetadata
}
export interface ProofCredentialAttributes {
  altCredentials?: string[]
  credExchangeRecord?: CredentialExchangeRecord
  credId: string
  credDefId?: string
  schemaId?: string
  credName: string
  attributes?: Attribute[]
}

export interface ProofCredentialPredicates {
  altCredentials?: string[]
  credExchangeRecord?: CredentialExchangeRecord
  credId: string
  credDefId?: string
  schemaId?: string
  credName: string
  predicates?: Predicate[]
}

export interface ProofCredentialItems extends ProofCredentialAttributes, ProofCredentialPredicates {}
