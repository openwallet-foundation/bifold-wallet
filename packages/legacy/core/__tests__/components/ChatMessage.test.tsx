import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'
import { MessageProps } from 'react-native-gifted-chat'

import { ChatMessage, CallbackType, ExtendedChatMessage } from '../../App/components/chat/ChatMessage'
import { Role } from '../../App/types/chat'
import { testIdWithKey } from '../../App/utils/testable'

const onDetailsMock = jest.fn()
const user = {
  _id: Role.me,
}
const currentMessage: ExtendedChatMessage = {
  _id: '1',
  user,
  messageOpensCallbackType: CallbackType.CredentialOffer,
  onDetails: onDetailsMock,
  renderEvent: jest.fn(),
  text: 'test',
  createdAt: new Date('2024-01-01'),
}
const props: MessageProps<ExtendedChatMessage> = {
  user,
  currentMessage: currentMessage,
  key: '1',
  position: 'left',
}

describe('ChatMessage', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Credential offer renders correctly', () => {
    props.currentMessage!.messageOpensCallbackType = CallbackType.CredentialOffer
    const tree = render(<ChatMessage messageProps={props} />)
    const button = tree.getByTestId(testIdWithKey('Chat.ViewOffer'))

    fireEvent(button, 'press')

    expect(onDetailsMock).toHaveBeenCalledTimes(1)
    expect(button).not.toBeNull()
    expect(tree).toMatchSnapshot()
  })

  test('Proof request renders correctly', () => {
    props.currentMessage!.messageOpensCallbackType = CallbackType.ProofRequest
    const tree = render(<ChatMessage messageProps={props} />)
    const button = tree.getByTestId(testIdWithKey('Chat.ViewRequest'))

    fireEvent(button, 'press')

    expect(onDetailsMock).toHaveBeenCalledTimes(1)
    expect(button).not.toBeNull()
    expect(tree).toMatchSnapshot()
  })

  test('Sent presentation renders correctly', () => {
    props.currentMessage!.messageOpensCallbackType = CallbackType.PresentationSent
    const tree = render(<ChatMessage messageProps={props} />)
    const button = tree.getByTestId(testIdWithKey('Chat.OpenPresentation'))

    fireEvent(button, 'press')

    expect(onDetailsMock).toHaveBeenCalledTimes(1)
    expect(button).not.toBeNull()
    expect(tree).toMatchSnapshot()
  })
})
