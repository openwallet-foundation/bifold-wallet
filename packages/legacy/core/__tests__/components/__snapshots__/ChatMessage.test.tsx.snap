// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage Credential offer renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "flex-end",
        "flexDirection": "column",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          },
          Object {
            "margin": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f0f0f0",
              "borderRadius": 15,
              "justifyContent": "flex-end",
              "marginRight": 60,
              "minHeight": 20,
            },
            null,
            null,
            Object {
              "backgroundColor": "#313132",
              "borderRadius": 4,
              "marginLeft": 0,
              "marginRight": 0,
              "padding": 16,
            },
          ]
        }
      >
        <View
          accessibilityRole="text"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "width": 50,
                }
              }
            >
              <
                height={40}
                width={40}
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#D3D3D3",
                    "fontSize": 12,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Jan 1, 2024, 12:00 am
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Chat.ViewOffer"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 32,
          "marginTop": 16,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
      testID="com.ariesbifold:id/Chat.ViewOffer"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        Chat.ViewOffer
      </Text>
    </View>
  </View>
</View>
`;

exports[`ChatMessage Proof request renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "flex-end",
        "flexDirection": "column",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          },
          Object {
            "margin": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f0f0f0",
              "borderRadius": 15,
              "justifyContent": "flex-end",
              "marginRight": 60,
              "minHeight": 20,
            },
            null,
            null,
            Object {
              "backgroundColor": "#313132",
              "borderRadius": 4,
              "marginLeft": 0,
              "marginRight": 0,
              "padding": 16,
            },
          ]
        }
      >
        <View
          accessibilityRole="text"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "width": 50,
                }
              }
            >
              <
                height={40}
                width={40}
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#D3D3D3",
                    "fontSize": 12,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Jan 1, 2024, 12:00 am
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Chat.ViewRequest"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 32,
          "marginTop": 16,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
      testID="com.ariesbifold:id/Chat.ViewRequest"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        Chat.ViewRequest
      </Text>
    </View>
  </View>
</View>
`;

exports[`ChatMessage Sent presentation renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "flex-end",
        "flexDirection": "column",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          },
          Object {
            "margin": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f0f0f0",
              "borderRadius": 15,
              "justifyContent": "flex-end",
              "marginRight": 60,
              "minHeight": 20,
            },
            null,
            null,
            Object {
              "backgroundColor": "#313132",
              "borderRadius": 4,
              "marginLeft": 0,
              "marginRight": 0,
              "padding": 16,
            },
          ]
        }
      >
        <View
          accessibilityRole="text"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "width": 50,
                }
              }
            >
              <
                height={40}
                width={40}
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#D3D3D3",
                    "fontSize": 12,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Jan 1, 2024, 12:00 am
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Chat.OpenPresentation"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 32,
          "marginTop": 16,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
      testID="com.ariesbifold:id/Chat.OpenPresentation"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        Chat.OpenPresentation
      </Text>
    </View>
  </View>
</View>
`;
